src/PID.cpp

```diff
@@ -10,12 +10,25 @@ PID::PID() {}
 
 PID::~PID() {}
 
-void PID::Init(double Kp, double Ki, double Kd) {
+void PID::Init(double kp, double ki, double kd) {
+  Kp = kp;
+  Ki = ki;
+  Kd = kd;
+
+  p_error = 0;
+  i_error = 0;
+  d_error = 0;
 }
 
 void PID::UpdateError(double cte) {
+  double prev_cte = p_error;
+
+  p_error = cte;
+  i_error += cte;
+  d_error = cte - prev_cte;
```

Comment:
nit: This could also be done with `p_error` directly before it's reset and remove the need for extra variable